Others/Dijkstra.java

```diff
@@ -104,7 +106,7 @@ public String toString() {
     /**
      * Builds a graph from a set of edges
      */
-    public Graph(Edge[] edges) {
+    public Graph(@Positive Edge[] edges) {

```

Comment:
The `@Positive` annotation is not in the NullnessChecker  hierarchy. You mixed the NullnessChecker and IndexChecker qualifiers in this case study. It is not wrong to annotate a file using 2 type systems, but it is not the usual way a case study is done.

Also, the annotation is incorrect. What you meant is that the array `edges` contains objects of type `Edge`, which can't be `@Positive`, because it is meaningless. Only types like `int`, which represent a number can be `@Positive`, `@NonNegative` and some other IndexChecker qualifiers.

Reply:
Sorry, I'll keep in mind next time.